File: front-end/website/src/api/law-service/articleApi.js
import axiosClient from "../axiosClient";

const urlLaw = "law/article";
const articleApi = {
    getByChapterId(chapterId) {
        const url = urlLaw + `/chapter/${chapterId}`;
        return axiosClient.get(url, { chapterId });
    },

    getTreeArticle(articleId) {
        const url = urlLaw + `/tree/${articleId}`;
        return axiosClient.get(url, { articleId });
    },
    getAllByPage(params) {
        const url = urlLaw + "/filter";
        return axiosClient.get(url, { params });
    },
};

export default articleApi;

File: front-end/mobile/lib/apis/articleApi.dart
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'package:mobile/constants/baseUrl.dart';


class ArticleApi {

  Future<dynamic> getByChapterId(String chapterId) async {
    final url = Uri.parse('$baseUrl/law-service/article/chapter/$chapterId');
    final response = await http.get(url);

    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load article by chapter ID');
    }
  }

  Future<dynamic> getTreeArticle(String articleId) async {
    final url = Uri.parse('$baseUrl/law-service/article/tree/$articleId');
    final response = await http.get(url);

    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load tree article');
    }
  }

  Future<dynamic> getAllByPage(Map<String, dynamic> params) async {
    final url = Uri.parse('$baseUrl/law-service/article/filter');
    final response = await http.get(url.replace(queryParameters: params));

    if (response.statusCode == 200) {
      return json.decode(response.body);
    } else {
      throw Exception('Failed to load articles by page');
    }
  }
}
